### PR TITLE
Expose activations for all layers in FeedFowardPolicy

### PR DESCRIPTION
**Pull request by kantneel**
Originally, this class exposed just the final latent layer. After these changes, `self.ff_out` will have activations from every layer kept together. This is done by changing `mlp_extractor`, since this will make it easier to expose all activations in other classes that may use `mlp_extractor` too.

### CODE REVIEW

File: stable_baselines/common/policies.py [hunk 1]
```diff
@@ -50,18 +50,23 @@ def mlp_extractor(flat_observations, net_arch, act_fun):
     :param net_arch: ([int or dict]) The specification of the policy and value networks.
         See above for details on its formatting.
     :param act_fun: (tf function) The activation function to use for the networks.
+    :param all_acts: (bool) whether to return lists of activations for each layer, or just the final activations.
```
Comment (AdamGleave) [inline stable_baselines/common/policies.py]: This parameter doesn't seem very useful to me: the user never calls `mlp_extractor` directly, and from inside Baselines I think we always call it with it set to `all_acts=True`?

I suggest we just get rid of it, and always consider the case of `all_acts=True`, which also simplifies this function. Note that this just extracts TensorFlow operations that, if run, will reveal the activations. We do not have to use them!

We can then change `step` in the `Transparent*Policy` classes to let us specify which activations we want to see instead?

**Review by AdamGleave — CHANGES_REQUESTED** [review]
Think get rid of `all_acts` parameter and simplify code, otherwise good

File: stable_baselines/common/policies.py [hunk 2]
```diff
@@ -434,7 +440,10 @@ def __init__(self, sess, ob_space, ac_space, n_env, n_steps, n_batch, reuse=Fals
...
                 pi_latent = vf_latent = cnn_extractor(self.processed_obs, **kwargs)
             else:
-                pi_latent, vf_latent = mlp_extractor(tf.layers.flatten(self.processed_obs), net_arch, act_fun)
+                pi_acts, vf_acts = mlp_extractor(tf.layers.flatten(self.processed_obs),
+                                                 net_arch, act_fun)
+                self.ff_out = {'value': vf_acts, 'policy': pi_acts}
+                pi_latent, vf_latent = pi_acts[-1], vf_acts[-1]
 
             self.value_fn = linear(vf_latent, 'vf', 1)
             self.ff_out = {'value': [vf_latent], 'policy': [pi_latent]}
```
Comment (AdamGleave) [inline stable_baselines/common/policies.py]: This is overwriting the previous `self.ff_out`?

**Review by AdamGleave — CHANGES_REQUESTED** [review]
Bug that causes only final layer to exposed

File: stable_baselines/common/policies.py (hunk 2 above)
Comment (kantneel) [reply stable_baselines/common/policies.py]: oops yeah I'll remove that. Should I maybe have a default value of self.ff_out, i.e. `None` that I assign to it beforehand?

Comment (AdamGleave) [reply stable_baselines/common/policies.py]: I'd suggest moving this assign to the first `if` branch (for if you call `cnn_extractor` instead)?  `None` would be awkward as you'd need to have users of this code special-case that.

Comment (kantneel) [reply stable_baselines/common/policies.py]: yeah that makes sense.

**Pull request merged**